lib/Dialect/ESI/runtime/cpp/lib/backends/Xrt.cpp

```diff
@@ -51,25 +51,40 @@ XrtAccelerator::connect(Context &ctxt, std::string connectionString) {
 }
 
 struct esi::backends::xrt::XrtAccelerator::Impl {
-  constexpr static char kernel[] = "esi_kernel";
+  constexpr static char kernel_name[] = "esi_kernel";
 
   Impl(std::string xclbin, std::string device_id) {
     if (device_id.empty())
       device = ::xrt::device(0);
     else
       device = ::xrt::device(device_id);
 
-    auto uuid = device.load_xclbin(xclbin);
-    ip = ::xrt::ip(device, uuid, kernel);
+    // Find memory group for the host.
+    ::xrt::xclbin xcl(xclbin);
+    std::optional<::xrt::xclbin::mem> host_mem;
+    for (auto mem : xcl.get_mems())
+      // The host memory is tagged with "HOST[0]". Memory type is wrong --
+      // reports as DRAM rather than host memory so we can't filter on that.
+      if (mem.get_tag().starts_with("HOST"))
+        host_mem = mem;
```

Comment:
Missing `break`? or some kind of handling if multiple `"HOST#` memories are found?

Reply:
Yeah, this probably deserves a check and exception.